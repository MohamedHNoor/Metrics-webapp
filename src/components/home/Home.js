import React, { useEffect, useState } from 'react';
import { useSelector, shallowEqual, useDispatch } from 'react-redux';
import { fetchCountry, searchCountry } from '../../redux/countries';
import AllCountries from '../all-countries/AllCounries';
import './home.styles.css';

let render = true;
const Home = () => {
  const country = useSelector((state) => state.countries, shallowEqual);
  const dispatch = useDispatch();
  const [searchField, setSearchField] = useState('');

  useEffect(() => {
    if (render) {
      dispatch(fetchCountry());
      render = false;
    }
  }, [dispatch]);

  useEffect(() => {
    if (searchField !== '') {
      setTimeout(() => {
        dispatch(searchCountry(searchField));
      }, 1000);
    } else {
      dispatch(fetchCountry());
    }
  }, [dispatch, searchField]);

  return (
    <div className="home-container">
      <div className="search-box">
        <img src="" alt="africa map" className="map-photo" />
        <div>
          <h3 className="africa-title">Africa</h3>
          <p className="africa-population">1.216 billion</p>
        </div>
        <input
          value={searchField}
          onChange={(e) => setSearchField(e.target.value)}
          type="search"
          placeholder="Search a country..."
          className="search"
        />
      </div>
      <div className="title-div">
        <h3 className="title">African Countries</h3>
      </div>
      <div className="list-page">
        <AllCountries data={country} />
      </div>
    </div>
  );
};

export default Home;
