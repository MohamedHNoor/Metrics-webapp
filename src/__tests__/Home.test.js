import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Home from '../components/home/Home';
import store from '../redux/configureStore';

it('should render the homepage components correctly', () => {
  const homepage = render(
    <BrowserRouter>
      <Provider store={store}>
        <Home />
      </Provider>
    </BrowserRouter>,
  );
  expect(homepage).toMatchSnapshot();
});
