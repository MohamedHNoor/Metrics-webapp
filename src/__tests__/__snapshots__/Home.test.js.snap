// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the homepage components correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home-container"
      >
        <div
          class="search-box"
        >
          <div
            class="continent-info"
          >
            <img
              alt="africa map"
              class="africa-map"
              src="Africa.svg"
            />
            <div>
              <h3
                class="africa-title"
              >
                Africa
              </h3>
              <p
                class="africa-population"
              >
                1216 billion People
              </p>
            </div>
          </div>
          <input
            class="search"
            placeholder="Search a country..."
            type="search"
            value=""
          />
        </div>
        <div
          class="title-div"
        >
          <h3
            class="title"
          >
            African Countries
          </h3>
        </div>
        <div
          class="list-page"
        >
          <div
            class="countries-container"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home-container"
    >
      <div
        class="search-box"
      >
        <div
          class="continent-info"
        >
          <img
            alt="africa map"
            class="africa-map"
            src="Africa.svg"
          />
          <div>
            <h3
              class="africa-title"
            >
              Africa
            </h3>
            <p
              class="africa-population"
            >
              1216 billion People
            </p>
          </div>
        </div>
        <input
          class="search"
          placeholder="Search a country..."
          type="search"
          value=""
        />
      </div>
      <div
        class="title-div"
      >
        <h3
          class="title"
        >
          African Countries
        </h3>
      </div>
      <div
        class="list-page"
      >
        <div
          class="countries-container"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
